chore: docs and changeset (#13870)
withastro/astro: .changeset/crazy-doors-buy.md

---
'astro': minor
---

> Adds the experimental support of Content Security Protection (CSP) in Astro. 

Reviewer: ```suggestion
Adds experimental support for Content Security Protection (CSP) in Astro. 
```

> CSP is an important feature for websites that require fine-grained control over the resources that can or cannot be downloaded and executed.

Reviewer: ```suggestion
CSP is an important feature to provide fine-grained control over resources that can or cannot be downloaded and executed by a document. In particular, it can help protect against [cross-site scripting (XSS)](https://developer.mozilla.org/en-US/docs/Glossary/Cross-site_scripting) attacks.
```

> With this new experimental feature, you can use most of Astro's features such as client islands, server islands, etc. and Astro will take care of everything.
> 
> The new feature has been designed to work in every environment out of the box (static pages, dynamic pages and single page applications), have maximum flexibility and type-safety in mind.

Reviewer: ```suggestion
Enabling this feature adds additional security to Astro's handling of processed and bundled scripts and styles by default, and allows you to further configure these, and additional, content types. This new experimental feature has been designed to work in every Astro rendering environment (static pages, dynamic pages and single page applications), while giving you maximum flexibility and with type-safety in mind.

It is compatible with most of Astro's features such as client islands, and server islands, athough Astro's view transitions using the `<ClientRouter />` are not yet fully supported. Inline scripts are not supported out of the box, but you can provide your own hashes for external and inline scripts.
```

---
'astro': minor
---

> Adds experimental support for Content Security Protection (CSP) in Astro. 

Reviewer 2: ```suggestion
Adds experimental Content Security Policy (CSP) support
```
